### PR TITLE
React Router Upgrade

### PR DESCRIPTION
**Pull request by minocys**
### CHANGES
1. Upgrade to React Router v4
    - Merge routes.js, Root.js, into AppVew
    - Move App components out of views folder
    - Replace browserHistory with history
    - Use withRouter HOC to pass history props to non-route components
    - Replace some Link components with NavLink component
    - Refactor Settings routes into Settings.jsx
    - Update index.jsx
1. Remove SSR
    - Delete renderer
    - Add and serve development and production html files
1. Refactor ImageLightBox logic out of AppView

### CODE REVIEW

**Comment by minocys** [comment]
@hibearpanda Download and test extensively plz before merge, if all good you should tag a release on both ws and ws-api so that ci will deploy

**Comment by minocys** [comment]
fuckin stupid yarn fuckin up my code +/- count fuck

File: public/views/Instance/index.jsx [hunk 1]
```diff
@@ -168,16 +167,17 @@ class Instance extends React.Component {
...
   openFeedback: PropTypes.func.isRequired,
-  fetchInstanceImages: PropTypes.func.isRequired,
-  concatImages: PropTypes.func.isRequired,
-  spliceImage: PropTypes.func.isRequired,
   openLightBox: PropTypes.func.isRequired,
-  clearImages: PropTypes.func.isRequired,
-  fetchInstanceDetails: PropTypes.func.isRequired,
-  clearInstanceDetails: PropTypes.func.isRequired,
-  params: PropTypes.object.isRequired
+  params: PropTypes.object.isRequired,
```
Comment (hibearpanda) [inline public/views/Instance/index.jsx]: No params obj - use match

File: public/views/InstanceRedirect.jsx [hunk 2]
```diff
@@ -22,7 +21,7 @@ class InstanceRedirect extends React.Component {
       fetch(instanceRoutes.fetchShortCode(this.props.params.id))
```
Comment (hibearpanda) [inline public/views/InstanceRedirect.jsx]: params -> match.params

File: public/App/AppView.jsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        auth={props.auth}
+        handleLogOut={props.handleLogOut}
+      />
+      <ModalContainer
+        auth={props.auth}
+        dismissModal={props.dismissModal}
+        modal={props.modal}
+      />
+      <Switch>
+        <Route exact path='/'  component={Landing} />
```
Comment (hibearpanda) [inline public/App/AppView.jsx]: Extra Space and use "" instead of ''

File: public/views/Settings.jsx [hunk 4]
```diff
@@ -29,9 +32,15 @@ function Settings(props) {
           links={links}
         />
         <div className="page-content-padding">
-          {React.cloneElement(props.children, {
-            auth: props.auth
-          })}
+          <Route 
+            exact
+            path='/'
```
Comment (hibearpanda) [inline public/views/Settings.jsx]: "" instead of ''

**Pull request merged**